### ISSUE TITLE
Catch invalid usernames in timelines

### ISSUE THREAD

**Issue by edsu** [DocNow/twarc#512]
The `timelines` command can throw a strange error if it encounters an invalid username. Is there a pattern we can use to check that it looks OK to throw at the API?

```
Traceback (most recent call last):
  File "C:\Users\s2989428\AppData\Local\Programs\Python\Python39\Scripts\twarc2-script.py", line 33, in <module>
    sys.exit(load_entry_point('twarc==2.3.10', 'console_scripts', 'twarc2')())
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\click\core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\click\decorators.py", line 38, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\twarc\command2.py", line 905, in timelines
    for response in tweets:
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\twarc\client2.py", line 137, in _search
    for response in self.get_paginated(url, params=params):
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\twarc\client2.py", line 780, in get_paginated
    resp = self.get(*args, **kwargs)
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\twarc\decorators2.py", line 88, in new_f
    raise e
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\twarc\decorators2.py", line 81, in new_f
    resp = f(self, *args, **kwargs)
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\twarc\decorators2.py", line 51, in new_f
    resp.raise_for_status()
  File "c:\users\s2989428\appdata\local\programs\python\python39\lib\site-packages\requests\models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.twitter.com/2/tweets/search/all?expansions=author_id%2Cin_reply_to_user_id%2Creferenced_tweets.id%2Creferenced_tweets.id.author_id%2Centities.mentions.username%2Cattachments.poll_ids%2Cattachments.media_keys%2Cgeo.place_id&user.fields=created_at%2Cdescription%2Centities%2Cid%2Clocation%2Cname%2Cpinned_tweet_id%2Cprofile_image_url%2Cprotected%2Cpublic_metrics%2Curl%2Cusername%2Cverified%2Cwithheld&tweet.fields=attachments%2Cauthor_id%2Ccontext_annotations%2Cconversation_id%2Ccreated_at%2Centities%2Cgeo%2Cid%2Cin_reply_to_user_id%2Clang%2Cpublic_metrics%2Ctext%2Cpossibly_sensitive%2Creferenced_tweets%2Creply_settings%2Csource%2Cwithheld&media.fields=duration_ms%2Cheight%2Cmedia_key%2Cpreview_image_url%2Ctype%2Curl%2Cwidth%2Cpublic_metrics&poll.fields=duration_minutes%2Cend_datetime%2Cid%2Coptions%2Cvoting_status&place.fields=contained_within%2Ccountry%2Ccountry_code%2Cfull_name%2Cgeo%2Cid%2Cname%2Cplace_type&query=from%3AAIHS_OHS%2F&max_results=100
```

**Comment by igorbrigadir** [comment]
Yeah, the usernames have to be alphanumeric, with a `_` and nothing else is allowed - so the regex for usernames would be: 

```
^(\w{1,15})$
```

(`\w` is `[a-zA-Z0-9_]` and username has to be between 1 and 15 chars)

**Comment by edsu** [comment]
This was fixed in dd37538aa5cbbbbf4fadc41f50bd44c2cf74e0cf
